mitodl/micromasters
# Added material-ui, replaced react-mdl Dialog with material-ui Dialog

#### What are the relevant tickets?
Required for #277

#### What's this PR do?
There's an issue with the material-ui AutoComplete because it uses z-index to hover above the textfield area. The HTML5 dialog element which react-mdl uses is defined as top-layer when activated using `dialog.showModal()` so we can't use it with the AutoComplete. Fortunately the material-ui Dialog is a drop in replacement

#### Where should the reviewer start?

#### How should this be manually tested?
Make sure the dialogs still work. There's one in App.js and one in the education tab when you click the edit or plus button. Cancel or ESC or clicking outside the dialog should cancel, and the other buttons should act like before.

#### Any background context you want to provide?

The react-select component will be truncated within the textbox, but we're getting rid of it in #277 anyway

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
